content/providers/05-observability/humanloop.mdx

```diff
@@ -0,0 +1,265 @@
+---
+title: Humanloop
+description: Monitor and trace your AI SDK application with Humanloop, the LLM evals platform for enterprises.
+---
+
+# Humanloop Observability
+
+[Humanloop](https://humanloop.com/) is the LLM evals platform for enterprises, giving you the tools that top teams use to ship and scale AI with confidence. Humanloop integrates with the AI SDK to provide:
+
+The AI SDK can log to [Humanloop](https://humanloop.com/) via OpenTelemetry. This integration enables trace visualization, cost/latency/error monitoring, and evaluation by code, LLM, or human judges.
+
+## Reference
+
+### Telemetry Configuration
+
+The AI SDK supports tracing through the `experimental_telemetry` parameter that can be set on each request.
+
+```ts
+const result = await generateText({
+  model: openai('gpt-4o'),
+  prompt: 'Write a short story about a cat.',
+  experimental_telemetry: { isEnabled: true },
+});
+```
+
+### Metadata Parameters
+
+The Humanloop OpenTelemetry Receiver accepts these metadata parameters:
+
+| Parameter             | Required | Description                                                                    |
+| --------------------- | -------- | ------------------------------------------------------------------------------ |
+| `humanloopPromptPath` | Yes      | Path to the prompt on Humanloop. Generation spans create Logs for this Prompt. |
+| `humanloopFlowPath`   | No       | Path to the flow on Humanloop. Groups steps into a single Flow Log.            |
+| `humanloopFlowId`     | No       | ID of a Flow Log on Humanloop. Groups multiple calls into a single Flow Log.   |
+
+## Setup
+
+### Prerequisites
+
+- A Humanloop account and API key.
+  - [Sign up](https://app.humanloop.com/signup) or [login](https://app.humanloop.com/login) to Humanloop.
+  - Create an API key in [Organization Settings](https://app.humanloop.com/account/api-keys).
+- A Vercel AI SDK application.
+
+### Telemetry Configuration
+
+When sending traces to Humanloop, these parameters are added to the telemetry object:
+
+```ts
+experimental_telemetry: {
+  isEnabled: true,
+  functionId: 'unique-function-id', // Optional identifier for the function
+  metadata: {
+    humanloopPromptPath: 'Path/To/Prompt',
+    humanloopFlowPath: 'Path/To/Flow', // Optional
+    humanloopFlowId: 'flow-log-id'     // Optional
+  },
+}
+```
+
+### Environment Variables
+
+When using OpenTelemetry with Humanloop, the following environment variables configure the OTLP exporter:
+
+```bash
+OTEL_EXPORTER_OTLP_ENDPOINT=https://api.humanloop.com/v5/import/otel
+OTEL_EXPORTER_OTLP_PROTOCOL=http/json
+OTEL_EXPORTER_OTLP_HEADERS="X-API-KEY=xxxxxx" # Humanloop API key
```

Comment:
Actually like the xxxxx pattern here; it's a good way to signal you need to add `X-API-KEY=` prefix